third-party/realdds/include/realdds/dds-stream-profile.h

```diff
@@ -83,18 +83,20 @@ class dds_stream_profile
 {
     dds_stream_uid _uid;
     dds_stream_format _format;
-    int16_t _frequency;  // "Frames" per second
+    int16_t _frequency; // "Frames" per second
+    int8_t _inner_type; // User defined type

```

Comment:
I don't quite understand the need here.
If really necessary, the "type" is the "stream type" and not the profile type. All profiles of the same stream will have the same type, no? I.e., it should be communicated with the stream and not the profile.

Reply:
RS applications like `rs-enumerate-devices` and `rs-sensor-control` use rs2_profile type field to print a "stream name" for each profile, and we need to pass this information to them.

I agree that this is a duplication that can be avoided by sending a stream type. Currently in notificaiton-msg it is sent in the profile struct, and I didn't want to change that so we won't have a difficult merge.

I will take the information to the stream from the first profile in the message assuming all profiles have the same type (true for libRS, not sure that true in the general case)

Reply:
So approve my PR and sending a type as part of the `stream-header` will be simple :)

Reply:
By the way -- I don't think the type should be an argument to the stream, but a subtype of the stream: i.e., a `dds_gyro_stream_server` derives from `dds_motion_stream_server`, etc...

Reply:
Then we will have to add types to realdds if LibRS adds a type. realdds doesn't really care about the type, it just need to relay it to the user

Reply:
But the type does change things, and can change functionality.
Right now it's divided into "video" and "motion" but if we need more, then more types are needed.
Doing this thru polymorphism is the right way to do it -- if it's important. I'm still unsure that it is (what depends on the stream type?) -- let's talk about it.